pandas/tests/indexes/test_base.py

```diff
@@ -707,6 +707,70 @@ def test_intersect_str_dates(self):
 
         assert len(res) == 0
 
+    @pytest.mark.parametrize(
+        'ftype, fname, stype, sname, expected_name',
+        [
+            ('copy', 'A', 'copy', 'A', 'A'),
+            ('copy', 'A', 'copy', 'B', None),
+            ('copy', 'A', 'copy', None, None),
+            ('copy', 'A', 'empty', 'A', 'A'),
+            ('copy', 'A', 'empty', 'B', None),
+            ('copy', 'A', 'empty', None, None),
+            ('copy', None, 'copy', 'B', None),
+            ('copy', None, 'copy', None, None),
```

Comment:
do the empties in a separate test from the copies, ok on parameterize the names though (on both tests)